### PR TITLE
Remove accents from country labels to match geo detected country data 

### LINKED ISSUE #38947: Geolocation warning shows up incorrectly
Reported pdibGW-26P-p2#comment-1585

### Description
For some locations, geolocation warning shows up when the detected country is correct. 

This is somewhat hard to reproduce as it requires VPN connection.

### Steps to reproduce

Download and install https://www.charlesproxy.com/ 

We'll need to make `https://www.charlesproxy.com/ ` return `{"latitude":"36.720172","longitude":"-4.420112","country_short":"ES","country_long":"Spain","region":"Andalucia","city":"Malaga"}` to fake our location

1. Open Charles Proxy and click Tools -> Rewrite
2. Check `Enable Rewrite` and add a new entry.
3. Add a new location and enter `https://public-api.wordpress.com/geo`
4. Add a new Type and Action
5. Change `Type` to Body
6. Add `{"latitude":"36.720172","longitude":"-4.420112","country_short":"ES","country_long":"Spain","region":"Andalucia","city":"Malaga"}` for `Replace` value

Screen Shot 2023-07-05 at 5 20 43 PM

1. Create a new JN site
2. Install and activate **[woocommerce-trunk-nightly.zip](https://github.com/woocommerce/woocommerce/releases/download/nightly/woocommerce-trunk-nightly.zip)**
3. Start the core profiler
4. Proceed to `Tell us a bit about your store` page
5. You should see a warning about your location.

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #38947  .

 The WP geo API returns regions without accents, while the country labels have accents. This PR removes accents from the country labels when comparing against geo-detected country to match them correctly.

### How to test the changes in this Pull Request:
This is somewhat tricky to test as you need to somehow make https://public-api.wordpress.com/geo/ endpoint return mock data. I used https://www.charlesproxy.com/ to do so.

If you're using Charles Proxy, 

1. Open Charles Proxy and click Tools -> Rewrite
2. Check `Enable Rewrite` and add a new entry.
3. Add a new location and enter `https://public-api.wordpress.com/geo`
4. Add a new Type and Action
5. Change `Type` to Body
6. Add `{"latitude":"36.720172","longitude":"-4.420112","country_short":"ES","country_long":"Spain","region":"Andalucia","city":"Malaga"}` for `Replace` value

Screen Shot 2023-07-05 at 5 20 43 PM

1. Checkout this branch in a new store
2. Start the core profiler
8. Proceed to `Tell us a bit about your  store` page
9. `Where is your store located?` should be autofilled and you should not see a warning.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement

#### Comment 

</details>

### CODE REVIEW

File: packages/js/onboarding/src/utils/countries/index.ts [hunk 1]
```diff
@@ -51,7 +51,14 @@ export const findCountryOption = (
...
 			option.label.includes( '—' )
 		) {
-			const wcRegion = option.label.split( '—' )[ 1 ].trim();
+			// WP GEO API Returns regions without accents.
+			// Remove accents from the region to compare.
+			// Málaga -> Malaga
+			const wcRegion = option.label
+				.split( '—' )[ 1 ]
+				.trim()
+				.normalize( 'NFKD' )
```
Comment (rjchow) [inline packages/js/onboarding/src/utils/countries/index.ts]: I think [localeCompare()](https://dev.to/mliakos/use-localecompare-to-ignore-string-case-and-diacritics-accents-1gbf) might be the most recent way to do this, with locale either undefined or 'en', and { sensitivity: base } ([docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator#parameters))

image

**Review by rjchow — COMMENTED** [review]
I think if there's a unit test for findCountryOption covering the accent comparison we can skip the manual testing since it's a bit convoluted 😃

File: packages/js/onboarding/src/utils/countries/index.ts (hunk 1 above)
Comment (moon0326) [reply packages/js/onboarding/src/utils/countries/index.ts]: Thank you for the suggestion! TIL 👍 I'll update it soon.

Comment (moon0326) [reply packages/js/onboarding/src/utils/countries/index.ts]: @rjchow updated in https://github.com/woocommerce/woocommerce/pull/39110/commits/fbd12eccdf831e20cba4fdc6bf71970fae18553c

**Review by chihsuan — APPROVED** [review]
LGTM and tested well! 💯

**Review by ilyasfoo — APPROVED** [review]
I wasn't able to use CharlesProxy, I had SSL issues with it.

I modified the response via JS instead. Tested well, LGTM! 🚢

File: packages/js/onboarding/src/utils/countries/index.ts (hunk 1 above)
Comment (rjchow) [reply packages/js/onboarding/src/utils/countries/index.ts]: thanks! looks good to me!

**Pull request merged**